packages/itwinui-css/backstop/tests/tabs.html

```diff
@@ -234,6 +249,20 @@ <h2>Default</h2>
             </button>
           </li>
         </ul>
+
+        <div class="iui-tabs-actions">
+          <button
+            class="iui-button iui-button-dropdown"
+            data-iui-size="small"
```

Comment:
The height of a default button is 36px whereas the height of tabs is 40px, so it should be possible to support the default size, not just small

Reply:
I tried this out before and saw that the vertical spacing between `iui-tabs-actions` and `iui-tabs-content` is too small.

![image](https://user-images.githubusercontent.com/45748283/210442435-8ba4a6a5-4948-4428-b720-143a5416a61e.png)

Also, this is similar to what you mentioned in https://github.com/iTwin/iTwinUI/issues/770#issuecomment-1260907982.

But if this small spacing is okay, then yes, we can support the `data-iui-size="normal"` also.

Reply:
i think in your screenshot it is center aligned, so the 4px distributed above and below. try aligning it to `start`

Reply:
Aligning it to `start` would almost work for the normal sized button. But for the small sized button, it seems quite misaligned vertically.

![image](https://user-images.githubusercontent.com/45748283/210592806-c712f4b5-60a5-472b-9a9a-9fe7c1757534.png)

Reply:
> i expect people will want to place normal size inputs and buttons in the actions area

I do agree with this.

We could increase the height by increasing the padding on tabs or on the right aligned actionable content area.  The one potential issue I see with including the padding on the actionable area is if the content inside appears / disappears based on what tab is active.  This would cause the tabs to grow / shrink in height as content is added / removed.

I'm playing around with it locally now, how about `min-height: calc(var(--iui-size-l) * 2);` on horizontal tabs?  The default button size is 36px & this would make the tabs 48px tall.
![Screenshot 2023-01-05 at 1 18 20 PM](https://user-images.githubusercontent.com/849817/210851972-cc18c896-4f4e-461e-8b5d-2ff25afe8423.png)
![Screenshot 2023-01-05 at 1 18 41 PM](https://user-images.githubusercontent.com/849817/210852065-32853491-a13e-47e6-8b40-16097535797d.png)

Reply:
I added a `min-height: calc(var(--iui-size-l) * 2);` to `.iui-tab` in the mixin `iui-tabs-horizontal`. [`47cb7f4` (#860)](https://github.com/iTwin/iTwinUI/pull/860/commits/47cb7f47cc5392d4afd10dc6951c252052d5429c)

---

> This would cause the tabs to grow / shrink in height as content is added / removed.

I think this would happen even when applying `min-height` to `.iui-tab`, because if `.iui-tabs-actions` even becomes larger than around `48px` when one of the tabs is selected, then `iui-tab` would match that height.

The below example is with the above mentioned `min-height` on `.iui-tab`. Here, clicking the tab named "Left" sets `data-iui-size="large"` on the dropdown button.

[Video7.webm](https://user-images.githubusercontent.com/45748283/210865807-78ba8eba-6646-49cd-8a9e-63faa7d4a7e4.webm)

But I feel adding a `min-height` to `.iui-tab` is at least more reliable than adding a padding to `.iui-tabs-actions`.

Reply:
1. What if we leave default tabs untouched? We can say it works best with small size inputs/buttons in the actions area. The reason: app ui widgets make heavy use of default tabs, so conserving space is probably important there.

2. For borderless tabs, we should find a standard gap between the line and the actions. Lets say 8px. So 36px (button height) + 8px (gap) =`44px` is the minimum height we need to set. We can still keep the actions center aligned vertically (instead of `start`), but ensure that there is always an 8px gap below the actions.

Reply:
> For borderless tabs, we should find a standard gap between the line and the actions. Lets say 8px. So 36px (button height) + 8px (gap) =44px is the minimum height we need to set. We can still keep the actions center aligned vertically (instead of start), but ensure that there is always an 8px gap below the actions.

To confirm, if we give a height of 44px to `.iui-tab`, then that gives 8px of free space for `.iui-tabs-actions`. But if `.iui-tabs-actions` is aligned centrally (vertically), then the gap between the bottom of the `.iui-tabs-actions` and the `.iui-tabs-content` is 4px, not 8px. Or am I misunderstanding something?

I was thinking of adding a bottom padding to `.iui-tabs-actions` in the `@iui-tabs-borderless` mixin. But I couldn't come up with a selector because `.iui-tabs-actions` and `.iui-tabs.iui-borderless` are siblings.

![image](https://user-images.githubusercontent.com/45748283/210884712-2645f4cb-d3d1-4227-b0ba-712e5bfb4db9.png)

Is there a better way to enforce a minimum of 8px of vertical spacing between `.iui-tabs-actions` and `.iui-tabs-content` when in borderless tabs?

Reply:
flexbox might be able to achieve this
![](https://user-images.githubusercontent.com/9084735/211062286-baad00d1-a423-492a-a705-7a168188c9c5.png)

Reply:
Thanks, used these ideas to make it work. I added the following spacing:

* For horizontal tabs: minimum actions-content gap of `var(--iui-size-3xs)` just so that actions will never stick to content ([L215](https://github.com/iTwin/iTwinUI/pull/860/files#diff-0dba596e6381aa5a0bd620f0bb73c94ef6e65fcb3ce323f61bb25704a27a16adR215))
<!-- ![image](https://user-images.githubusercontent.com/45748283/211111855-a56ebc5e-a866-4fa0-b904-889ed3449a66.png) -->
![image](https://user-images.githubusercontent.com/45748283/211112034-3005df14-8203-4c1b-b8e3-4b72a276e58c.png)

* For horizontal borderless tabs: minimum actions-content gap of `var(--iui-size-xs)` (`8px`) ([L220](https://github.com/iTwin/iTwinUI/pull/860/files#diff-0dba596e6381aa5a0bd620f0bb73c94ef6e65fcb3ce323f61bb25704a27a16adR220))
(Because of the min spacing, the actions button is slightly above the horizontal centerline)
![image](https://user-images.githubusercontent.com/45748283/211111917-3daf50a7-9bec-497d-a9a4-05d3d62270f8.png)

* For vertical tabs: horizontal padding of `var(--iui-size-xs)` ([L216](https://github.com/iTwin/iTwinUI/pull/860/files#diff-0dba596e6381aa5a0bd620f0bb73c94ef6e65fcb3ce323f61bb25704a27a16adR216))
![image](https://user-images.githubusercontent.com/45748283/211111978-995e8f6d-6a5c-4d74-91b3-955fbd68716d.png)

Are these spacings alright? @mayank99 @FlyersPh9

Reply:
what was wrong with my suggested approach? the minimum height of the horizontal tabs has now increased to 48px (it was 44px in my suggestion) because there is an extra 4px above the actions.